### ISSUE TITLE
App Crashes in iOS 10.3.2 iPhone

### ISSUE THREAD

**Issue by ChirayuShah** [fechanique/cordova-plugin-fcm#348]
When I open the app it shows the splash screen. Then it asks for "Allow Notification" prompt. Then the app crashes.

The error in logs shows that "abort()" is called.

The app crashes also after opening again.

What should I do.

**Comment by 7doctor7** [comment]
I changed Cordova iOS platform version to 4.3.0 
It's worked for me.

**Comment by matko238** [comment]
Use iOS platform >= 4.3.0 and update cordova (i use cordova 6.5 and ios platform 4.4.0

**Comment by Sroose** [comment]
I'm using platform 4.4.0 and cordova 6.5.0 but have the exact same issue since installing this plugin

**Comment by sparklton** [comment]
Same here - cordova 7.0.1, ios platform 4.4.0, the plugin crashes on 10.1.1 iPhone

**Comment by sparklton** [comment]
OK, I have found what is wrong here...
Looking into the ipa I see that the GoogleService-info.plist is empty.
Digging around I also see that the directory structure created by the build has duplicate Resources folders:

1. /platforms/ios/myapp/Resources/GoogleService-Info.plist
2. /platforms/ios/myapp/Resources/Resources/GoogleService-Info.plist

The first one contains correct plist file I provided, the second is empty - which eventually gets placed into the ipa and that makes the application kaput. Getting rid of #2 folder doesn't help. What did help is manually placing content of the first into second. Then ipa contains correct resource and application work fine.

I have a question at this point:

It is obvious that this plugin was released without ever being tried or tested for iOS. 
Is there any way to prevent crash of the entire application when plugin fails, isn't it possible to catch the exception and just continue without it? My knowledge of writing plugins fails me here. But in my case if this plugin doesn't work, it is perfectly normal to continue with the rest of the app. Yet crashing the end user without a warning is a huge show stopper.

**Comment by cyclonmaster** [comment]
I have same problem. Gonna try sparklton solution.
Cordova 7.0.1, Cordova iOS 4.4.0 iOS 10.3.3.
Change to Cordova 4.3.1 also not help.
Change to cordova-plugin-fcm 1.1.4 also crash.

**Comment by amit777** [comment]
as a short term hack, I had to copy the plist files manually:
```
cp GoogleService-Info.plist platforms/ios/MyApp/Resources/Resources/GoogleService-Info.plist
cp GoogleService-Info.plist platforms/ios/MyApp/Resources/GoogleService-Info.plist

```

**Comment by SkateFreak** [comment]
i'm getting errors both for ios and android.

1. ios: the app builds, but crashes
2. android: the build fails

what can i do if i can't manually editing the files? (using PhoneGap Build)

**Comment by davidvexel** [comment]
Thank you @sparklton, had the same problem but got fixed with your solution.

**Comment by nicovanzyl** [comment]
Will this be fixed in the official plugin?

**Comment by DouglasLapsley** [comment]
I have the same issue.  Totally new Ionic app with just this plugin.  Works perfectly on a build for Android, crashes after the permission screen on iOS build.  Cordova for iOS v4.5.4.  I don't know of a way to edit the IPA to implement @sparklton's solution.  

Anyone found a solution yet?

**Comment by davidvexel** [comment]
@DouglasLapsley Yeah you need to use @sparklton's solution, it's really easy, just need copy the file from `/platforms/ios/myapp/Resources/GoogleService-Info.plist`  to `/platforms/ios/myapp/Resources/Resources/GoogleService-Info.plist`

**Comment by DouglasLapsley** [comment]
Thanks David.  I am sure this will appear a thick question, but it sounds like I need to edit the contents of the IPA itself.  How do I do that please?  I am using Ionic Pro Cloud build rather than XCode.  Thanks :)

**Comment by davidvexel** [comment]
No worries, you don't really need to do it in Xcode, just go to your project folder and look for the `platforms` directory, then copy this file I mentioned, you can do that on Finder if you're using mac.

**Comment by DouglasLapsley** [comment]
Hmmm, thanks, but no luck.  

I tried copying the files as mentioned before committing to Ionic for building.  That didn't work.  I think that Ionic Pro ignores the platforms/ folders and recreates them server side.  

I also tried editing the IPA directly by changing it to a .zip and editing the GoogleService-Info.plist in there.  That just resulted in a provisioning error (no surprise there as I believe you need to resign the IPA after making modifications in some way).

Gonna have a look at OneSignal, which I understand to be pretty decent too.

Thanks!
